node/src/components/storage.rs

```diff
@@ -556,7 +556,25 @@ impl Storage {
         match incoming.message {
             NetRequest::Deploy(ref serialized_id) => {
                 let id = decode_item_id::<Deploy>(serialized_id)?;
-                let opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+                let mut opt_item = self.get_deploy(id).map_err(FatalStorageError::from)?;
+
+                match &mut opt_item {
+                    Some(deploy) => {
+                        let mut txn = self
+                            .env
+                            .begin_ro_txn()
+                            .expect("could not create RO transaction");
```

Comment:
I saw this kind of panic invoked by a few functions specifically around opening transactions in this module, but perhaps we should remove them all. Maybe this could be a tech debt ticket and not addressed in this PR?